index.js

```diff
@@ -10,9 +10,93 @@ function ask(questionText) {
 start();
 
 async function start() {
-  console.log("Let's play a game where you (human) make up a number and I (computer) try to guess it.")
-  let secretNumber = await ask("What is your secret number?\nI won't peek, I promise...\n");
-  console.log('You entered: ' + secretNumber);
-  // Now try and complete the program.
+
+  // Define constant range to start game:
+  let min = 1;
+  let max = 101;
+
+  //Greet user and offer game rules:
+  console.log("Hello. I am Lieutenant Commander Data.\nPlease think of a number between 1 and 100 (inclusive).\nI will try to guess it.")
+
+  //Gather user response if they have thought of a number within the range; display response for information:
+  let randNumResp = await ask("Have you selected a random number between 1 and 100? (Y/N)\n");
+
+  console.log('You entered: ' + randNumResp);
+
+  if (randNumResp === 'N') {
+    console.log("I'll assume you've chosen by now.");
+
+  };
+
+  //Declare and initialize app-level variables:
+  let highLow;
+  let middle;
+  let lastGuess;
+  let nextGuess = Math.floor((max + min) / 2);
+  let tryCount = 1;
+  // Ask user if guess is correct; first run will be for range set in app-level variables:
+  let userChoice = (await ask("Is it... " + nextGuess + "? (Y/N)\n")).toUpperCase();
+
+  ////////////////////////////////////////////
+
+  // Declare function guessNum() to calculate binary search:
+  /*
+    pivot is equal to the number known to not be the correct answer within the range.
+    option is the user input as to whether the game should guess higher or lower.
+    middle is set to be the midpoint in the range defined by the pivot and either the max or min as defined prior
+  */
+  function guessNum(pivot, option) {
+
+    if (option === 'H') {
+      middle = Math.floor((max + pivot) / 2);
+    } else if (option === 'L') {
+      middle = Math.floor((pivot + min) / 2);
+    };
+    return middle;
+  }
+
+  ////////////////////////////////////////////////
+
+
+  // Start while loop to generate successive guesses:
+  while (userChoice !== 'Y') {
+
+
+    if (userChoice === 'N') {
+
+      highLow = (await ask('Hmmm...is it higher(H) or lower(L)?')).toUpperCase();
+
+      if (highLow === 'H') {  // evaluate the user's highLow input for meaning 'go higher' or 'go lower'
+        min = nextGuess;      // set the value of min to that of the prior guess
+      } else if (highLow === 'L') {
+        max = nextGuess;     // set the value of max to that of the prior guess
+      };
```

Comment:
When setting the next min and max values, you should adjust by one for each. If the computer guesses 25 and the user says higher, then the computer shouldn't be able to guess 25 again, the minimum number the computer should be able to guess is now 26 ( or guess + 1). This is true for the minimum and maximum, except the max should be adjusted by -1.